### ISSUE TITLE
We build too many modes by default

### ISSUE THREAD

**Issue by espindola** [scylladb/scylla#4568]
With most build systems each build has exactly one mode. In scylla, the default build configuration builds debug, dev and release.

That is an odd default, as each build serves different purposes and each user probably wants only one of them.

We should probably change our build so that
* configure.py still produces a build.ninja that can handle all modes
* running just "ninja" either prints an error or defaults to only one mode.
* test.py either requires --mode or --mode defaults to a single mode.

**Comment by denesb** [comment]
I think when deciding on this we should consider somebody just wanting to compile scylla (without necessarily knowing much about its internals).
I think we should make the following work-flow work well:
```
./configure.py
ninja
```
I propose that `configure.py`'s `--mode` defaults to `release` and the default target of `ninja` should just compile the `scylla` executable. We could perhaps have a separate `ninja test` target that would compile and run the tests with `tests.py`.

**Comment by espindola** [comment]
I agree on configure.py having only one mode by default, but I thought there was disagreement on that, which is why I suggested changing just ninja first.
